spec.bs

```diff
@@ -111,26 +117,44 @@ Each [=environment settings object=] has an <dfn for="environment settings objec
+    1. [=list/Append=] the element to the element's <a spec="html">node document</a>'s [=Document/list of pending import map scripts=].
+    1. [=Fetch an import map=] given <var ignore>url</var>, <var ignore>settings object</var>, and <var ignore>options</var>.
     1. When the previous step asynchronously completes with |result|:
-       1. Decrement |settings object|'s [=environment settings object/pending import maps count=].
-          <p class="note">If this decreases the [=environment settings object/pending import maps count=] to 0, it will (asynchronously) unblock any [=wait for import maps=] algorithm instances.</p>
-       1. [=Register an import map=] given |result|, |settings object|, |base URL|, and the element.
+       1. If |result| is a failure, then set [=HTMLScriptElement/the script's import map=] to null.
+       1. Otherwise, set [=HTMLScriptElement/the script's import map=] to the result of [=parsing an import map string=], given |result| and |base URL|.
+       1. [=The import map is ready=].
     1. Return.
 - Insert the following before <a spec="html">prepare a script</a> step 25.2:
   - If <a spec="html">the script's type</a> is "`importmap`", then:
-    1. [=Register an import map=] given <var ignore>source text</var>, |settings object|, |base URL|, and the element.
+    1. [=list/Append=] the element to the element's <a spec="html">node document</a>'s [=Document/list of pending import map scripts=].
+    1. Set [=HTMLScriptElement/the script's import map=] to the result of [=parsing an import map string=], given <var ignore>source text</var> and |base URL|.
+    1. [=The import map is ready=].
     1. Return.
 
 <p class="issue">CSP is applied to import maps just like JavaScript scripts. Is this sufficient? <a href="https://github.com/WICG/import-maps/issues/105">#105</a>.</p>
 
 <p class="note">For import maps, the script never becomes <a spec="html" lt="the script is ready">ready</a> and <a spec="html">the script's script</a> remains null.</p>
 
+<div algorithm="the import map is ready hook">
+  When [=HTMLScriptElement/the import map is ready=], run the following steps:
+
+  1. If the element is not the first element in [=Document/list of pending import map scripts=] to which it was added above, then return without executing the script yet.
+  1. <em>Register</em>: [=/Register an import map=] given the first element of [=Document/list of pending import map scripts=].
+  1. Remove the first element of [=Document/list of pending import map scripts=].
+     <p class="note">If this makes the [=Document/list of pending import map scripts=] empty, it will (asynchronously) unblock any [=wait for import maps=] algorithm instances.</p>
+  1. If the [=Document/list of pending import map scripts=] is still not empty and the first entry's [=HTMLScriptElement/the import map is ready=], then jump back to the step labeled <em>Register</em>.
```

Comment:
Modern spec style is to use loops instead of gotos. Maybe tweak this to use a loop instead, if it's not too much work.

Reply:
Sounds totally fine. (I just copy-pasted the corresponding structure of the existing spec). I'll update.

Reply:
Done.